orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+import os
+import re
+import shelve
+import warnings
+import itertools
+from datetime import datetime
+
+import numpy as np
+from Bio import Entrez
+from Bio import Medline
+from validate_email import validate_email
+
+from Orange.canvas.utils import environ
+from Orange.data import StringVariable, DiscreteVariable, Domain
+from orangecontrib.text.corpus import Corpus
+
+PUBMED_TEXT_FIELDS = [
+    ('authors', 'FAU'),
+    ('article_title', 'TI'),
+    ('mesh_headings', 'MH'),
+    ('abstract', 'AB'),
+    ('pub_date', 'DP'),
+]
+
+
+def _join_if_list(value):
+    if isinstance(value, list):
+        return ' '.join(value)
+    return value
+
+
+def _mesh_headings_to_class(mesh_headings):
+    """Extract a class from the mesh headings.
+
+    We take the first mesh heading and extract it to use as a class.
+
+    Args:
+        mesh_headings (str): The string containing all the mesh headings.
+
+    Returns:
+        str: The class value.
+    """
+    # e.g. heading1/heading2,heading3/*heading4
+    try:
+        regex = re.compile(r'^(.\w*?)\W')
```

Comment:
This does not work. Some examples:
* `['Abscisic Acid/pharmacology', 'Cryopreservation/*methods']` -> `Abscisic` (should be `Abscisic Acid`)
* `['Animals', 'Bees/classification/*genetics/physiology']` -> `None` (should be `Animals`)

Also, I do not quite get the regex. What would the `.` supposed to capture? Why is the `\w` set to non-gready? Why should there always be a non-alfanumeric at the end? The example above show that this is not always the case.